lib/mayaUsd/ufe/UsdShaderNodeDef.cpp

```diff
@@ -361,6 +363,12 @@ Ufe::InsertChildCommand::Ptr UsdShaderNodeDef::createNodeCmd(
     TF_AXIOM(fShaderNodeDef);
     UsdSceneItem::Ptr parentItem = std::dynamic_pointer_cast<UsdSceneItem>(parent);
     if (parentItem) {
+        if (parentItem->nodeType() == "Scope"
+            && parentItem->nodeName()
+                == UsdUndoAssignNewMaterialCommand::resolvedMaterialScopeName()) {
+            return UsdUndoAddNewMaterialCommand::create(
+                parentItem, fShaderNodeDef->GetIdentifier());
+        }
```

Comment:
[Note] This implicit material creation can probably be removed again in the future. There is a UFE interface for material creation in the works and once it's done, it should be able to take care of this.